src/tools/shapes/CircleCreator.ts

```diff
@@ -18,32 +19,45 @@ export default class CircleCreator extends Shape implements CircleCreatorInterfa
     this.circle = new Konva.Circle()
   }
 
-  create = (canvasElement: CanvasElement, layer: Konva.Layer): void => {
+  create = (canvasElement: CanvasElement, layer: Konva.Layer, event: CustomEvent | CustomStageEvent): void => {
     this.group = new Konva.Group()
     this.group.attrs.temporary = this.temporary
     if (this.showRadius && canvasElement.tool.showRadius) {
       layer.add(this.group.id(canvasElement.id).add(
-        this.circle = this.createCircleElement(canvasElement),
-        this.line = this.createLineElement(canvasElement)
+        this.circle = this.createCircleElement(canvasElement, event),
+        this.line = this.createLineElement(canvasElement, event)
       ))
     } else {
       layer.add(this.group.id(canvasElement.id).add(
-        this.circle = this.createCircleElement(canvasElement)
+        this.circle = this.createCircleElement(canvasElement, event)
       ))
     }
   }
   // eslint-disable-next-line
-  move = (canvasElement: CanvasElement, layer: Konva.Layer, pos: Position): void => {
-    this.circle.radius(this.calcRadius(canvasElement.data[0], canvasElement.data[1], pos.x, pos.y))
+  move = (canvasElement: CanvasElement, layer: Konva.Layer, pos: Position, event: CustomEvent | CustomStageEvent): void => {
+    this.circle.radius(
+      this.calcRadius(
+        this.formatX(canvasElement.data[0], event),
```

Comment:
Canvaselements are stored in the state with global positions, and are formated when creating them

Reply:
The format functions are written inside class Shape which all tool extend. All tools can access these functions